.gitpod.yml

```diff
@@ -1,35 +1,48 @@
 # Learn more about this file at https://www.gitpod.io/docs/references/gitpod-yml
 tasks:
   - name: Service Server
-    # Get Web API urls for use in Next build
+    # Copying env.development and replacing 'localhost' wit Gitpod urls
+    # Passing Gitpod urls for use in Next build
     before: |
-      eval $(gp env -e GITPOD_API_URL=$(gp url 3000))
-      eval $(gp env -e GITPOD_WEB_URL=$(gp url 8000))
-      eval $(gp env -e GITPOD_MICROSERVICE_URL=$(gp url 8443))
+      eval $(gp env -e API_URL=$(gp url 3000))
+      eval $(gp env -e WEB_URL=$(gp url 8000))
+      eval $(gp env -e API_HOST=$(gp url 8443))
+
+      sed -r \
+          -e  "s@(.+=)http://localhost:8000(/[^ ]*)*@\1$WEB_URL\2@g" \
+          -e  "s@(.+=)http://localhost:3000(/[^ ]*)*@\1$API_URL\2@g" \
+          -e  "s@(.+=)http://localhost([^:]*)@\1$API_HOST\2@g" \
+          -e  "s@development\.yml@gitpod\.yml@" \
+          config/env.development > .env
     init: |
       npm install
-      docker-compose --env-file config/env.gitpod build
-    # GitPod workspace urls are not stable, we have to look up urls and build again
+      docker-compose --env-file .env build
+    # GitPod workspace urls are not stable, we'd like to force rebuild nginx again with new urls
     command: |
-      docker-compose --env-file config/env.gitpod up -d
-      npm run start
+      docker-compose --env-file .env up -d --no-dep nginx
+      docker-compose --env-file .env up -d
+      gp await-port 9200 && npm run start
```

Comment:
Add a comment here that we're waiting on Elasticsearch, so it's clear why you're doing this.  Having said that, I'm not clear if we need this.  @manekenpix has code in our Elasticsearch client that retries connections.

Reply:
On some occasions, the `npm run start` script will cancel itself before Elasticsearch has a chance to start and be connected to. I don't know if the retries are based on a certain number of attempts or after a certain amount of time. For some reason, we usually need to wait longer for Elasticsearch

Reply:
We should probably file an issue to bump the number of retries we do, see https://github.com/Seneca-CDOT/telescope/blob/29951b7b60fa33d83e14b05f24d4a4c2722f9ef7/src/backend/utils/indexer.js#L127-L167

Reply:
The back-end uses [this](https://github.com/Seneca-CDOT/telescope/blob/e1675d3a963cc787032c518cd1ebe208e8e4e0ea/config/env.development#L237) to determine for how long it should try to connect to ES (it's time, not number of attempts). If it can't connect, the back-end exits, **BUT** in production/staging we run the back-end in a container with [unless-stopped](https://github.com/Seneca-CDOT/telescope/blob/e1675d3a963cc787032c518cd1ebe208e8e4e0ea/docker/production.yml#L35), so it tries again after shutting down.